motor/core.py

```diff
@@ -1670,6 +1773,37 @@ class AgnosticCommandCursor(AgnosticBaseCursor):
 
     _CommandCursor__die = AsyncRead()
 
+    def _lazy_init(self):
+        if not self.delegate:
+            self.delegate = self._target.delegate.watch(**unwrap_kwargs_session(self._kwargs))
+
+    def _try_next(self):
+        # This method is run on a thread.
+        self._lazy_init()
+        return self.delegate.try_next()
+
+    async def __try_next(self):
+        """Advance the cursor without blocking indefinitely.
+
+        This method returns the next document without waiting
+        indefinitely for data.
+
+        If no document is cached locally then this method runs a single
+        getMore command. If the getMore yields any documents, the next
+        document is returned, otherwise, if the getMore returns no documents
+        (because there is no additional data) then ``None`` is returned.
+
+        :Returns:
+          The next document or ``None`` when no document is available
+          after running a single getMore or when the cursor is closed.
+        """
+        loop = self.get_io_loop()
+        return await self._framework.run_on_executor(loop, self._try_next)
+
+    # TODO: MOTOR-1169
+    if hasattr(CommandCursor, "try_next"):
+        try_next = __try_next
```

Comment:
Could you add a manual test for this? Eg a test that uses cursor_command and then calls try_next?

Reply:
Done